src/stransport_stream.c

```diff
@@ -110,20 +110,38 @@ int stransport_certificate(git_cert **out, git_stream *stream)
 	return 0;
 }
 
+/*
+ * Contrary to typical network IO callbacks, Secure Transport write callback is
+ * expected to write *all* passed data, not just as much as it can, and any
+ * other case would be considered a failure.
```

Comment:
This is not surprising in libgit2. Every write method will write everything or return a failure. It is surprising on the read side, but it's business as usual when writing.